websites/migrations/0001_initial.py

```diff
@@ -0,0 +1,79 @@
+# Generated by Django 3.1 on 2020-12-14 20:34
+
+from django.db import migrations, models
+import django.db.models.deletion
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = []
+
+    operations = [
+        migrations.CreateModel(
+            name="Website",
+            fields=[
+                ("created_on", models.DateTimeField(auto_now_add=True)),
+                ("updated_on", models.DateTimeField(auto_now=True)),
+                (
+                    "uuid",
+                    models.UUIDField(
+                        default=uuid.uuid4,
+                        primary_key=True,
+                        serialize=False,
+                        unique=True,
+                    ),
+                ),
+                ("title", models.CharField(max_length=512)),
+                ("publish_date", models.DateTimeField(blank=True, null=True)),
+                ("type", models.CharField(default="course", max_length=24)),
+                ("metadata", models.JSONField(blank=True, null=True)),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+        migrations.CreateModel(
+            name="WebsiteContent",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("created_on", models.DateTimeField(auto_now_add=True)),
+                ("updated_on", models.DateTimeField(auto_now=True)),
+                ("uuid", models.UUIDField(default=uuid.uuid4)),
```

Comment:
Should this be the primary key ?   is it 'id' necessary ?

Reply:
@Ferdi I think it may be possible for content files in different OCW courses to have the same uuid.  For example, in open-discussions-rc right now there are 143,812 OCW content files but only 120,883 distinct uid's.

Reply:
Here is an example of a course image, with uid '35b03ee5439763ef0fab624b90fc8601', used in 2 courses, with different S3 URLs:

https://open-learning-course-data-rc.s3.amazonaws.com/5-72-statistical-mechanics-spring-2012/35b03ee5439763ef0fab624b90fc8601_5-72s12.jpg

https://open-learning-course-data-rc.s3.amazonaws.com/5-72-non-equilibrium-statistical-mechanics-spring-2012/35b03ee5439763ef0fab624b90fc8601_5-72s12.jpg

Reply:
sure, and it's important to support that feature - resource reuse between courses

Reply:
@gumaerc what do you think, does it make sense for content uuid and site (ie course) to be unique together for this model, but not uuid alone?  Another option could be making this a ManyToMany relation instead with a unique uuid (one content object, for multiple courses), but fields like `markdown` and `metadata` might differ between courses.